### PR TITLE
Add sql query to get full category list and generate filter html

### PR DESCRIPTION
**Pull request by CloudNiner**
greenworks1

### CODE REVIEW

File: app/scripts/cartodb/categories-service.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function Categories ($log, $q, Config) {
+
+        var categories = null;
+        var categoriesQuery = [
+            'SELECT COUNT(*), SECTION, project_category, sub_category FROM master_datalist',
```
Comment (flibbertigibbet) [inline app/scripts/cartodb/categories-service.js]: Nit to pick: `SECTION` should be lower case in the query (column name, not SQL keyword).

File: app/scripts/cartodb/categories-service.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            'GROUP BY SECTION, project_category, sub_category',
+            'ORDER BY SECTION, project_category, sub_category'
```
Comment (flibbertigibbet) [inline app/scripts/cartodb/categories-service.js]: Doesn't the default ordering match the group by columns? Being explicit is fine, though.

File: app/scripts/views/map/map-controller.js [hunk 3]
```diff
@@ -16,8 +16,44 @@
...
+            ctl.onSubFilterClicked = onSubFilterClicked;
+
             cartodb.createVis('map', 'https://greenworks.cartodb.com/api/v2/viz/33627780-f988-11e4-8ff7-0e0c41326911/viz.json')
                 .done(onVisReady, onVisError);
+
+            Categories.get().then(function (categories) {
+                ctl.categories = categories;
+                ctl.filters =
+
+                // Cheat and delay showing filters for one more digest cycle
```
Comment (flibbertigibbet) [inline app/scripts/views/map/map-controller.js:38]: Well that's awkward.

**Comment by flibbertigibbet** [comment]
+1

File: app/scripts/cartodb/categories-service.js (hunk 2 above)
Comment (CloudNiner) [inline app/scripts/cartodb/categories-service.js]: I believe so.

File: app/scripts/views/map/map-controller.js (hunk 3 above)
Comment (CloudNiner) [inline app/scripts/views/map/map-controller.js:38]: :turtle: :turtle: :turtle:

**Pull request merged**